lib/perl/Genome/InstrumentData/Command/AlignAndMerge.pm

```diff
@@ -0,0 +1,175 @@
+            is => 'HASH',
+            doc => 'mapping of labels to user objects. Will be added to any generated results',
+        },
+    ],
+    has_optional_input => [
+        bedtools_version => {
+            is => 'Text',
+            doc => 'The version of Bedtools to use when needed by aligners/filters',
+        },
+        annotation_build_id => {
+            is => 'Number',
+            doc => 'Id of the annotation build to use when aligning',
+            is_optional => 1,
+        },
+        instrument_data_segment_id => {
+            is => 'Text',
+            doc => 'A specific segment of the data to align',
+        },
+        instrument_data_segment_type => {
+            is => 'Text',
+            doc => 'How the data is segmented',
+        },
+        instrument_data_filter => {
+            is => 'Text',
+            valid_values => [ 'forward-only', 'reverse-only', undef ],
+        },
+        force_fragment => {
+            is => 'Boolean',
+            default_value => 0,
+            doc => 'Treat all reads as fragment reads',
+        },
+        trimmer_name => {
+          is => 'Text',
+          doc => 'name of the read trimmer to use before alignment is performed'
+        },
+        trimmer_params => {
+          is => 'Text',
+          doc => 'params to use with the read trimmmer'
+        },
+        trimmer_version => {
+          is => 'Text',
+          doc => 'version of the read trimmer to use'
+        },
+    ],
+    has_optional_output => [
+        result_id => {
+            is => 'Text',
+            doc => 'The ID of the result generated/found when running the command',
+        },
+        alignment_result => {
+            is => 'Genome::InstrumentData::AlignmentResult::Merged',
+            id_by => 'result_id',
+            doc => 'The result generated/found when running the command',
+        }
+    ],
+};
+
+sub execute {
+    my $self = shift;
+
+    my $result = $self->_process_alignments('get_or_create');
+    unless ($result) {
+        $self->error_message("Error finding or generating alignments!");
+        return 0;
+    }
+    $self->result_id($result->id);
+
+    my $per_lane_results = $self->_process_per_lane_alignments('get_or_create');
+    unless ($per_lane_results) {
+        $self->error_message("Error finding or generating per-lane alignments!");
+        return 0;
+    }
+
+    return $result;
+}
+
+sub shortcut {
+    my $self = shift;
+
+    my $result = $self->_process_alignments('get_with_lock');
+    unless ($result) {
+        return undef;
+    }
+    $self->result_id($result->id);
+
+    my $per_lane_results = $self->_process_per_lane_alignments('get_with_lock');
+    unless ($per_lane_results) {
+        return undef;
+    }
+
+    return $result;
+}
+
+sub _process_alignments {
+    my $self = shift;
+    my $mode = shift;
+
+    my %params = $self->_alignment_params;
+    my $result = Genome::InstrumentData::AlignmentResult::Merged::Speedseq->$mode(
```

Comment:
I pushed a commit to create the class names more generically from the `name` param.